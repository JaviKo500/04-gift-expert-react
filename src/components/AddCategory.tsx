import { useState } from "react";

interface Props {
  onNewCategory: (newCategory: string) => void;
}

export const AddCategory = ( { onNewCategory }: Props ) => {
  const [inputValue, setInputValue] = useState('');
  const onInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setInputValue(event.target.value);
  }
  const onSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    if( inputValue.trim().length <= 1) return;
    onNewCategory(inputValue.trim());
    setInputValue('');
  }
  return(
    <>
      <h5>Add a category</h5>
      <form onSubmit={onSubmit}>
        <input type="text" 
          placeholder="Search gift" 
          value={inputValue}
          onChange={onInputChange}
          />
      </form>
    </>
  );
}