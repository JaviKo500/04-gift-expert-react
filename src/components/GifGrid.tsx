import { useEffect, useState } from "react";
import { getGifts } from "../helpers/getGifts";
import type { GiftInterface } from "../interfaces/gift.interface";

export const GifGrid = ({ category }: { category: string }) => {
  //* rebuild the gif's when the category changes
  const [gifs, setGifts] = useState<GiftInterface[]>([]);
  const getNewGifs = async (category: string) => {
    const newGifts = await getGifts(category);
    setGifts(newGifts);
  }
  useEffect(
    () => {
      getNewGifs(category);
    },
    [ category ]
  );
  if ( gifs.length ) {
    return (
      <>
        <h2>{category}</h2>
        <ol>
          {
            gifs.map( ( { id, title } ) => {
              return <li key={id}>{title}</li>
            })
          }
        </ol>
      </>
    );  
  }
  return (
    <>
      <div>
        <h2>{category}</h2>
        <p>Loading...</p>
      </div>
    </>
  );
}