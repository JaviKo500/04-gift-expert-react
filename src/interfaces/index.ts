export * from './gif-api.interface';
export * from './gif.interface';