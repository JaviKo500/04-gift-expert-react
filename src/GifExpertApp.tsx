import { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {
  const [categories, setCategories] = useState<string[]>([]);

   const onAddCategory = ( newCategory: string ) => {
    if( categories.includes( newCategory ) ) return;
    setCategories( [ newCategory, ...categories ] );
  }

  return (
    <>
      <h4>Gif Expert</h4>
      <AddCategory onNewCategory={ onAddCategory } />
      {
        categories.map( ( category ) => {
          return <GifGrid key={category} category={ category } />
        })
      }
    </>
  );
}
