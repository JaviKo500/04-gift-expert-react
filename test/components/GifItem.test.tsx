import { render } from '@testing-library/react';
import { GifItem } from '../../src/components';

describe('GifItem.test', () => {
  const title = 'title';
  const url = 'https://example.com/image.gif';
  test( 'should match snapshot', () => {
    const { container } = render( <GifItem title={title} url={url} /> );
    expect( container ).toMatchSnapshot();
  });
});