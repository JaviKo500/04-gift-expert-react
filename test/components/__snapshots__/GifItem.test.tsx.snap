// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem.test should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="title"
      src="https://example.com/image.gif"
    />
    <p>
      title
    </p>
  </div>
</div>
`;
